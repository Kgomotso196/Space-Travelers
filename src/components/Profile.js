import React from 'react';

const Profile = () => (
  <div>
    <h1>My Profile</h1>
  </div>
);

export default Profile;
