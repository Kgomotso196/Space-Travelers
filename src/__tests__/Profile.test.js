import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Profile from '../components/Profile';
const mockStore = configureMockStore();
describe('Profile Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      missions: {
        missions: [
          {
            mission_id: 'mission1',
            mission_name: 'Mission 1',
            joined: true,
          },
          {
            mission_id: 'mission2',
            mission_name: 'Mission 2',
            joined: true,
          },
        ],
      },
      rockets: {
        rockets: [
          {
            id: 'rocket1',
            name: 'Rocket 1',
            reserved: true,
          },
          {
            id: 'rocket2',
            name: 'Rocket 2',
            reserved: true,
          },
        ],
      },
    });
  });
  test('renders Profile component correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Profile />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});