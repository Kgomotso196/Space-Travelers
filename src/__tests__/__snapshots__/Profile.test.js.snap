// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component renders Profile component correctly 1`] = `
<DocumentFragment>
  <div
    id="profile-container"
  >
    <div
      class="list"
    >
      <h2>
        My Missions
      </h2>
      <ul
        id="missions"
      >
        <li>
          Mission 1
          <button>
            Leave
          </button>
        </li>
        <li>
          Mission 2
          <button>
            Leave
          </button>
        </li>
      </ul>
    </div>
    <div
      class="list"
    >
      <h2>
        My Rockets
      </h2>
      <ul
        id="rockets"
      >
        <li>
          Rocket 1
          <button>
            Cancel
          </button>
        </li>
        <li>
          Rocket 2
          <button>
            Cancel
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
